### ISSUE TITLE
Support promises

### ISSUE THREAD

**Issue by devinivy** [devinivy/haute-couture#1]
The hapi API supports promises, so I think haute-couture should too!  Looking for usage allowed like this,
```js
const HauteCouture = require('haute-couture');

module.exports = (server, options, next) => {

  HauteCouture()(server, options)
  .then(() => {

    // Custom plugin code
    next();
  })
  .catch(next);
};

module.exports.attributes = {
  name: 'my-hapi-plugin'
};
```

**Comment by WilliamSWoodruff** [comment]
Challenge accepted

**Comment by devinivy** [comment]
Do as much work as you'd like to do!  Someone could pick-up where you left off.  But this will be complete when we have code, tests, and docs 👍

**Comment by WilliamSWoodruff** [comment]
Cool, thanks!

**Comment by devinivy** [comment]
PR in the works at #2.

**Comment by devinivy** [comment]
Closed in #2
